### PR TITLE
New circle ci docs

### PR DESCRIPTION
**Pull request by lakkeger**
# Motivation
We'd like to provide a drop-in snippet library for the users in circle-ci too.

# Changes
- Add startup snippet
- Add config section
- Add pro activation part
- Add snippet for log dumping
- Store state variations (workspace, cache, cloud pods, ephemeral instance)

### CODE REVIEW

File: content/en/user-guide/ci/circle-ci/index.md [hunk 1]
```diff
@@ -3,77 +3,437 @@ title: "CircleCI"
...
+      ...
+      - localstack/wait
+workflows:
+  localstack-test:
+    jobs:
+      - localstack-test
+```
+
+### Configuration
+To configure localstack use `environment` key on the job level or a shell command, where the latter takes higher precedence.
```
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:52]: nit:
```suggestion
To configure localstack use the `environment` key on the job level or a shell command, where the latter takes higher precedence.
```

File: content/en/user-guide/ci/circle-ci/index.md
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:1]: q: This page - similar to other CI docs - contain some references to "LocalStack Pro+" which I think should be changed to something like "LocalStack Starter+"? /cc @HarshCasper

File: content/en/user-guide/ci/circle-ci/index.md [hunk 2]
```diff
@@ -3,77 +3,437 @@ title: "CircleCI"
...
+
+Read more about the config options [here](/references/configuration/).
```
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:54]: nit: 
```suggestion
Read more about the config options of LocalStack [here](/references/configuration/).
```

File: content/en/user-guide/ci/circle-ci/index.md [hunk 3]
```diff
@@ -3,77 +3,437 @@ title: "CircleCI"
...
+    steps:
+      - run:
+          name: Configure Localstack
+          command: echo 'export DEBUG=1' >> "$BASH_ENV"
+...
+```
+
+### Configuring a CI key
+
+To enable LocalStack Pro+, you need to add your LocalStack CI API key to the project's environment variables.
```
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:84]: nit:
```suggestion
To enable LocalStack Pro+, you need to add your LocalStack CI key to the project's environment variables.
```

File: content/en/user-guide/ci/circle-ci/index.md
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:1]: This page contains 21 references to "Localstack" (with a lower letter s in stack), while the official naming is "LocalStack" (with a capital S in Stack). Would be great to adjust that across the page.

File: content/en/user-guide/ci/circle-ci/index.md [hunk 4]
```diff
@@ -3,77 +3,437 @@ title: "CircleCI"
...
+Cloud pods providing an easy solution to persist Localstack's state, even between workflows or projects.
 
-            echo "Waiting for LocalStack startup..."  
-            localstack wait -t 30                     
-            echo "Startup complete"
-            
+Find more information about cloud pods [here](/user-guide/state-management/cloud-pods/).
+
+##### Multiple projects
+Update cloud pod in it's own project (ie IaC repo).
```
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:224]: > Update cloud pod in it's own project (ie IaC repo).

I am not 100% sure what "ie IaC repo" means. Maybe you could phrase this in a more verbose fashion?

File: content/en/user-guide/ci/circle-ci/index.md [hunk 5]
```diff
@@ -3,77 +3,437 @@ title: "CircleCI"
...
+          command: localstack pod save <POD_NAME>
+
+
+workflows:
+  localstack-build:
+    jobs:
+      - localstack-update-cloud-pod
+```
+
+In a separate project use the "golden" cloud pod as below:
```
Comment (alexrashed) [inline content/en/user-guide/ci/circle-ci/index.md:253]: > In a separate project use the "golden" cloud pod as below:

What's a golden cloud pod?

**Review by alexrashed — COMMENTED** [review]
Awesome! Great to have the tested collection of useful snippets now for CircleCI as well! 🚀 💯 

I only found a few nitpicks, some typos, and some questions on capitalization and tier naming. I will leave the final signoff to @HarshCasper, since he did the review of the other CI pages.

File: content/en/user-guide/ci/circle-ci/index.md (hunk 5 above)
Comment (lakkeger) [reply content/en/user-guide/ci/circle-ci/index.md:253]: The term "golden" comes from the AWS Golden AMI concept ([here](https://aws.amazon.com/blogs/awsmarketplace/announcing-the-golden-ami-pipeline/) or [here](https://docs.aws.amazon.com/managedservices/latest/appguide/ex-immute-gold-ami.html)), which is a preconfigured AMI or in this case cloud pod. The source of it is irrelevant here, only important that it exists and available for usage without any necessary configuration or deployment.

File: content/en/user-guide/ci/circle-ci/index.md (hunk 4 above)
Comment (lakkeger) [reply content/en/user-guide/ci/circle-ci/index.md:224]: Here it means that the infrastructure code is stored in a separate repo which runs its own pipeline that creates/updates the cloud pod we want to use here.

File: content/en/user-guide/ci/circle-ci/index.md (hunk 5 above)
Comment (alexrashed) [reply content/en/user-guide/ci/circle-ci/index.md:253]: Okay, would it make sense to define this term before using it? Maybe something like 'create a "golden" cloud pod' in the section where the pod is created?

File: content/en/user-guide/ci/circle-ci/index.md (hunk 4 above)
Comment (alexrashed) [reply content/en/user-guide/ci/circle-ci/index.md:224]: That makes sense, could you maybe explain this in the docs / make the sentence a bit more verbose?

File: content/en/user-guide/ci/circle-ci/index.md [hunk 6]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
 linkTitle: "CircleCI"
 weight: 4
 description: >
-  Use LocalStack in CircleCI
+  Use LocalStack in Circle CI
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:6]: The official product name is "CircleCI" https://circleci.com/

File: content/en/user-guide/ci/circle-ci/index.md [hunk 7]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
...
 ---
 
+This page contains easily customisable snippets to show you how to manage LocalStack in a [Circle CI](https://circleci.com) workflow.
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:9]: No text is required before intro :)

File: content/en/user-guide/ci/circle-ci/index.md [hunk 8]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
...
+    docker:
+      - image: cimg/base:2024.02
+    steps:
+      - run:
+        command: |
+          response=$(curl -X POST -d '{"auto_load_pod": "false"}' \
+            -H 'ls-api-key: $LOCALSTACK_API_KEY' \
+            -H 'authorization: token $LOCALSTACK_API_KEY' \
+            -H 'content-type: application/json' \
+            https://api.localstack.cloud/v1/previews/my-gitlab-state)
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:376]: Gitlab?

File: content/en/user-guide/ci/circle-ci/index.md [hunk 9]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
...
+          command: localstack pod save <POD_NAME>
+
+
+workflows:
+  localstack-build:
+    jobs:
+      - localstack-update-cloud-pod
+```
+
+In a separate project use the previously created "golden" cloud pod as below:
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:260]: What is a "golden" cloud pod? Also the product name is `Cloud Pod`

File: content/en/user-guide/ci/circle-ci/index.md [hunk 10]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
...
-            localstack wait -t 30                     
-            echo "Startup complete"
-            
+#### Cloud Pods
+Cloud pods providing an easy solution to persist LocalStack's state, even between workflows or projects.
+
+Find more information about cloud pods [here](/user-guide/state-management/cloud-pods/).
+
+##### Multiple projects
+Update or create the cloud pod in it's own project (ie in a separate Infrastructure as Code repo), this would create a base/"golden" cloud pod, which you can use in the future without any configuration or deployment.
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:231]: Golden Cloud Pod is still a loosely defined term used internally. Lets not use it on public-facing docs

File: content/en/user-guide/ci/circle-ci/index.md [hunk 11]
```diff
@@ -3,77 +3,444 @@ title: "CircleCI"
...
+More information about state management [here](/user-guide/state-management/export-import-state).
 
-            echo "Waiting for LocalStack startup..."  
-            localstack wait -t 30                     
-            echo "Startup complete"
-            
+#### Cloud Pods
+Cloud pods providing an easy solution to persist LocalStack's state, even between workflows or projects.
+
+Find more information about cloud pods [here](/user-guide/state-management/cloud-pods/).
```
Comment (HarshCasper) [inline content/en/user-guide/ci/circle-ci/index.md:228]: Avoid adding hyperlinks in unclear words like `here`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**